src/SAWScript/HeapsterBuiltins.hs

```diff
@@ -163,17 +164,31 @@ lookupFunctionDecl :: LLVMModule arch -> String -> Maybe L.Declare
 lookupFunctionDecl lm nm =
   find ((fromString nm ==) . L.decName) $ L.modDeclares $ modAST lm
 
+-- | Look up the 'L.Define' for a symbol defined in an 'LLVMModule'
+lookupFunctionDef :: LLVMModule arch -> String -> Maybe L.Define
+lookupFunctionDef lm nm =
+  find ((fromString nm ==) . L.defName) $ L.modDefines $ modAST lm
```

Comment:
We've gone from a `Map` lookup to a list lookup, which seems somewhat unfortunate. Could this be a performance bottleneck in practice?

Reply:
I'm not sure; I don't have a good sense of how hot this lookup function is. I'd be surprised if it is a major bottleneck, but it's a shame to be going backward in terms of algorithmic complexity.

Ideally, the llvm-pretty library, or some intermediate layer would provide better lookup functionality for this stuff so it doesn't have to be redone everywhere downstream.

Reply:
@eddywestbrook, would you be willing to take a quick look at these heapster-releated changes and give a sense of whether this is a concern for you?